instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex

```diff
@@ -145,6 +156,18 @@ defmodule OpentelemetryPhoenix do
     |> OpenTelemetry.Ctx.attach()
   end
 
+  @doc false
+  def handle_socket_dispatch_start(_event, _measurements, meta, config) do
+    if config.adapter == :cowboy2, do: cowboy2_start()
```

Comment:
This was necessary as it didn't update the actual span. I assume the difference from the router event is that the socket handling happens much earlier than router, and there is some handling of cowboy process in phoenix that links it correctly.